test/Test/Database/LSMTree/Internal/Merge.hs

```diff
@@ -159,19 +159,8 @@ prop_MergeUnion fs hbio stepSize (SmallList rds) =
   where
     rds' = fmap serialiseRunData rds
 
-    rhs :: Map SerialisedKey (SerialisedValue, Maybe SerialisedBlob)
-    rhs = Map.unionsWith resolveValueAndBlob
-            (map (Map.mapMaybe getValueAndBlob . unRunData) rds')
-
-    getValueAndBlob :: Entry.Entry v b -> Maybe (v, Maybe b)
-    getValueAndBlob = \case
-        Entry.Insert v           -> Just (v, Nothing)
-        Entry.InsertWithBlob v b -> Just (v, Just b)
-        Entry.Mupdate v          -> Just (v, Nothing)
-        Entry.Delete             -> Nothing
-
-    resolveValueAndBlob (v', Nothing) (v, b) = (mappendValues v' v, b)
-    resolveValueAndBlob (v', Just b)  (v, _) = (mappendValues v' v, Just b)
```

Comment:
The reason I think this can be simplified is because we are doing almost the same as in `prop_MergeDistributes`, but we're also testing the semantics of `combineUnion`. We can test the semantics of `combineUnion` separately. What do you think?

Reply:
e.g., see `prop_resolveEntriesNormalSemantics`: we could do something similar for the `combineUnion` semantics

Reply:
That's true. I could add a property test like that, or do you want to do it?

I also think we could remove the whole `prop_MergeUnion`. It doesn't add much over the existing `prop_MergeDistributes`.

Reply:
I'll add the property test now